### ISSUE TITLE
Ask: how to manage sysroot librairies?

### ISSUE THREAD

**Issue by Lyvsain** [conan-io/conan#5915]
Hello ! This is more a question of how to do it than an issue. 

Conan version: **1.8.1**

I'm working on a compagny network which is not connected to the internet so we manage our own Artifactory repository. We have a set of open sources software (OSS), on the shelve (OTS, compagny software), application (APP) and several platforms (PT) on which we cross-compile these softwares. 

Here is a summary description of how everything is connected:

```
oss_a: v1.0.0, v2.0.0 

ots_a      v1.0.0                       v1.0.1
          [requires]                    [requires]
          oss_a/1.0.0                   oss_a/2.0.0

ots_b:    v1.0.0 
          [requires]
          ots_a/[~1.x]

app_a:                                  app_b:
          [requires]                    [requires] 
          ots_b/1.0.0                   ots_b/1.0.0
          ots_a/1.0.1                   ots_a/1.0.0
```
**PT**
```
pt_a: contains oss_a v1.0.0
pt_b: contains oss_a v2.0.0
```

Now, we see that pt_a and pt_b have each one version of oss_a in their sysroot. So when we cross-build for this platform, we **do not want** to use a package from Conan but this in our sysroot/toolchain. 
We tried to add a key `platform.provided_version` in our profiles so when we `create export-pkg` we add or not the requirement but it messed up the sha generation and I end up with `missing prebuilt package` when I try to `conan install`.

Does anyone got a tip ?

**Comment by memsharded** [comment]
Hi @Lyvsain 

Just to make sure, are you using Conan 1.8.1, or 1.18.1? Because it is possible that things have improved since 1.8.1. Could you please try with the latest Conan?

Also it is not very clear what are the platforms that you describe in your issue, could you clarify? 
Similar thing with ``platform.provided_version``, I don't fully understand how you use it. Is it a custom setting, or an option to the package?

Maybe trying to set up a repo with a small repro case would really help. Recipes can actually be empty, and a small script that calls the sequence of commands. If you could provide it, great. Otherwise, don't worry, just please try to add a bit more of detail and clarify the above issues. Thanks!

**Comment by Lyvsain** [comment]
Hi @memsharded !

Yeap sorry, it's **1.18.1**.
Both platforms are Yocto based x86 Linux, an old one and new one. 

`platform_provided_version` is an option we added in our profiles. I added a zip file recreating a local conan cache and adding sample example, follow README.md to execute what we do.

[test-conan.zip](https://github.com/conan-io/conan/files/3733789/test-conan.zip)

Then, do conan have a way to manage package that are already present on the sysroot ?

**Comment by memsharded** [comment]
Hi again @Lyvsain 

Now that I see the code I have another bunch of questions, sorry 😄 

- The ``oss_a`` package doesn't really have requirements or build requirements, so it could be simplified right?
- I see the ``"platform_provided_version": "ANY"`` option in the ``oss_a`` package, but I fail to see how it is used in the binary. The idea is that such option should be used in the build or something, so different values of the option will provide different implementations. How is this option intended to be used within the recipe?
- Something similar for the ``platform`` setting. Not clear how it is used in the recipe, if used at all. Have you considered poky-linux to be a subsetting (``distro`` subsetting of the ``os``) instead?
- The ``oss_a`` recipe will not build with the ``CMake()`` helper, because that typically needs a ``settings.compiler`` defined. I guess it is just an example.
- What is the relation between the ``platform.sdkVersion=2.0.0`` and the ``oss_a:platform_provided_version=1.0.0`` settings that you are for example defining in a profile?

Are you in the CppLang slack? Because we are sometimes there, it might be useful and faster to discuss some of these things in a chat there. Thanks!

**Comment by memsharded** [comment]
We have added ``[platform_requires]`` new feature in Conan 2.0 to represent dependencies that are in the system, and also ``[replace_requires]`` to create wrapper recipes to model system requirements.

I am closing this as implemented in 2.0, but please create new tickets for any further issue or question you might have, thanks!
